flink-runtime/src/test/java/org/apache/flink/runtime/io/disk/BatchShuffleReadBufferPoolTest.java

```diff
@@ -0,0 +1,286 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/** Tests for {@link BatchShuffleReadBufferPool}. */
+public class BatchShuffleReadBufferPoolTest {
+
+    @Rule public Timeout timeout = new Timeout(60, TimeUnit.SECONDS);
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIllegalTotalBytes() {
+        createBufferPool(0, 1024);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIllegalBufferSize() {
+        createBufferPool(BatchShuffleReadBufferPool.MIN_TOTAL_BYTES, 0);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIllegalRequestTimeout() {
+        createBufferPool(0);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testInitializeAfterDestroy() {
+        BatchShuffleReadBufferPool bufferPool = createBufferPool();
+        bufferPool.destroy();
+        bufferPool.initialize();
+    }
+
+    @Test
+    public void testLargeTotalBytes() {
+        BatchShuffleReadBufferPool bufferPool = createBufferPool(Long.MAX_VALUE, 1024);
+        assertEquals(Integer.MAX_VALUE, bufferPool.getNumTotalBuffers());
+        bufferPool.destroy();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testTotalBytesSmallerThanBufferSize() {
+        BatchShuffleReadBufferPool bufferPool = createBufferPool(4096, 32 * 1024);
+        try {
+            bufferPool.initialize();
+        } finally {
+            bufferPool.destroy();
+        }
+    }
+
+    @Test
+    public void testBufferCalculation() {
+        long totalBytes = BatchShuffleReadBufferPool.MIN_TOTAL_BYTES;
+        for (int bufferSize = 4 * 1024; bufferSize <= totalBytes; bufferSize += 1024) {
+            BatchShuffleReadBufferPool bufferPool =
+                    createBufferPool(BatchShuffleReadBufferPool.MIN_TOTAL_BYTES, bufferSize);
+
+            assertEquals(totalBytes, bufferPool.getTotalBytes());
+            assertEquals(totalBytes / bufferSize, bufferPool.getNumTotalBuffers());
+            assertTrue(bufferPool.getNumBuffersPerRequest() <= bufferPool.getNumTotalBuffers());
+            assertTrue(bufferPool.getNumBuffersPerRequest() > 0);
+        }
+    }
+
+    @Test
+    public void testRequestBuffers() throws Exception {
+        BatchShuffleReadBufferPool bufferPool = createBufferPool();
+        List<MemorySegment> buffers = new ArrayList<>();
+
+        try {
+            buffers.addAll(bufferPool.requestBuffers(this));
+            assertEquals(bufferPool.getNumBuffersPerRequest(), buffers.size());
+        } finally {
+            bufferPool.recycle(buffers, this);
+            assertEquals(bufferPool.getNumTotalBuffers(), bufferPool.getAvailableBuffers());
```

Comment:
this check looks irrelevant to this test case. Can we remove it and add a separate test case `testRecycle` to test the normal simple cycle case, because currently I can only see some complex recycle test cases?